src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -44,31 +44,26 @@ class CoreWorkerDirectActorTaskSubmitter {
   /// Submit a task to an actor for execution.
   ///
   /// \param[in] task The task spec to submit.
+  /// \param[in] rpc_client The gRPC client to the actor. This may be null if
+  /// we do not know where the actor is yet.
   /// \return Status::Invalid if the task is not yet supported.
-  Status SubmitTask(const TaskSpecification &task_spec);
+  Status SubmitTask(const TaskSpecification &task_spec,
+                    std::shared_ptr<rpc::DirectActorClient> &rpc_client);
 
-  /// Handle an update about an actor.
-  ///
-  /// \param[in] actor_id The ID of the actor whose status has changed.
-  /// \param[in] actor_data The actor's new status information.
-  void HandleActorUpdate(const ActorID &actor_id, const gcs::ActorTableData &actor_data);
-
- private:
-  /// Push a task to a remote actor via the given client.
-  /// Note, this function doesn't return any error status code. If an error occurs while
-  /// sending the request, this task will be treated as failed.
+  /// Create connection to actor and send all pending tasks.
+  /// Note that this function doesn't take lock, the caller is expected to hold
+  /// `mutex_` before calling this function.
```

Comment:
EXCLUSIVE_LOCKS_REQUIRED(mu_)

Reply:
https://github.com/abseil/abseil-cpp/blob/master/absl/base/thread_annotations.h#L112

Reply:
I think this was only true before. I changed this function to hold a lock.